bazel/BUILD

```diff
@@ -299,6 +299,14 @@ config_setting(
     },
 )
 
+selects.config_setting_group(
```

Comment:
Hey @lizan can you do a pass of just the bazel magic?

Reply:
cc @keith also

Reply:
As long as this works it's definitely fine, I'm impressed it does though, we should separately change the apple config to use this instead, looking at the skylib impl it's very hard to parse, but I believe it's doing that same hack https://github.com/bazelbuild/bazel-skylib/blob/f80bc733d4b9f83d427ce3442be2e07427b2cc8d/lib/selects.bzl#L145-L186

Reply:
Believe it or not, I found this trick by googling, where I found the bazel issue you opened about the apple hack, where the person commented it could be done with this thing, but it wasn't merged yet. Now it's merged. :)

Reply:
Amazing, I forgot this, here's the update for the other ones that can benefit from this https://github.com/envoyproxy/envoy/pull/15797